source/vision/providerBase.py

```diff
@@ -7,35 +7,131 @@
+	"""
+	Base class for settings for a vision enhancement provider.
+	Ensure that the following are implemented:
+	- AutoSettings.getId:
+			This is case sensitive. Used in the config file. Does not have to match the module name.
+	- AutoSettings.getTranslatedName:
+			The string that should appear in the GUI as the name
+	- AutoSettings._get_supportedSettings:
+			The "runtime" settings for your provider
+	Although technically optional, derived classes probably need to implement:
+	- AutoSettings._get_preInitSettings:
+			The settings always configurable for your provider
+	"""
+	supportedSettings: SupportedSettingType  # Typing for autoprop L{_get_supportedSettings}
+
+	def __init__(self):
+		super().__init__()
+		self.initSettings()  # ensure that settings are loaded at construction time.
+
+	@classmethod
+	def _getConfigSection(cls) -> str:
+		return "vision"  # all providers should be in the "vision" section.
+
+
+class VisionProviderStateControl:
+	""" Stub showing the interface for controlling the start/termination of a single provider.
+			Implementors of this class should handle the outcome when things go wrong.
 	"""
 
-	_configSection = "vision"
+	@abstractmethod
+	def startProvider(self) -> None:
+		"""Initializes the provider in a way that is GUI friendly,
+		showing an error if appropriate.
+		@note: Use getProviderInstance to determine success
+		"""
+
+	@abstractmethod
+	def terminateProvider(self, verbose: bool = False) -> None:
+		"""Terminates one or more providers in a way that is GUI friendly,
+		@verbose: Whether to show a termination error.
+		@note: Use getProviderInstance to determine success
+		"""
+
+	@abstractmethod
+	def getProviderInstance(self):
+		"""Gets an instance of the provider if it already exists
+		@rtype: Optional[VisionEnhancementProvider]
+		"""
+
+	@abstractmethod
+	def getProviderInfo(self):
+		"""
+		@return: The provider info
+		@rtype: providerInfo.ProviderInfo
+		"""
+
+
+class VisionEnhancementProvider(AutoPropertyObject):
+	"""A class for vision enhancement providers.
+	Derived classes should implement:
+	- terminate:
+			How to shutdown the provider
+	- registerEventExtensionPoints:
+			Allows the provider to receive updates form NVDA
+	- canStart:
+			Checks startup dependencies are satisfied
+	- getSettings:
+			Returns your implementation of VisionEnhancementProviderSettings
+	Optional: To provide a custom GUI, return a SettingsPanel class type from:
+	- getSettingsPanelClass
+	"""
 	cachePropertiesByDefault = True
-	#: Override of supportedSettings to be a class property.
-	supportedSettings = ()
-	#: The roles supported by this provider.
-	#: This attribute is currently not used,
-	#: but might be later for presentational purposes.
-	supportedRoles: FrozenSet[Role] = frozenset()
-
-	def reinitialize(self):
-		"""Reinitializes a vision enhancement provider, reusing the same instance.
+
+	@classmethod
+	def getSettings(cls) -> VisionEnhancementProviderSettings:

```

Comment:
Do we want to prescripe that a provider has settings? IN that case, I think this method should be abstract:

```suggestion
	@abstractmethod
		def getSettings(cls) -> VisionEnhancementProviderSettings:
```